src/core/solutions/brown_function.cpp

```diff
@@ -0,0 +1,45 @@
+#include <math.h>
+#include <core/solutions/brown_function.h>
+
+namespace dnn_opt
+{
+namespace core
+{
+namespace solutions
+{
+    brown_function* brown_function::make(generator* generator, unsigned int size)
+    {
+        auto* result = new brown_function(generator, size);
+        
+        result->init();
+        
+        return result;
+    }
+    float brown_function::calculate_fitness()
+    {
+        float result = 0;
+        float* params = get_params();
+        
+        solution::calculate_fitness();
+        
+        for(int i = 0;i < size() - 1; i++)
+        {
+           float res1=pow(params[i], 2);
+           float res2=pow(params[i+1], 2);

```

Comment:
Review spaces between operators: `=`  and `+` signs.

Performance improvement suggestion: here we find the square of the current position and the next one, what if we store the square of the previous position and reuse the calculation in the next iteration?

This way, we only have to calculate the square of a single number instead of two numbers in a single iteration...